src/react/hooks/use-presence-listener.tsx

```diff
@@ -0,0 +1,178 @@
+   */
+  readonly presenceData: PresenceMember[];
+
+  /**
+   * Provides access to the underlying {@link Presence} instance of the room.
+   */
+  readonly presence: Presence;
+}
+
+/**
+ * A hook that provides access to the {@link Presence} instance in the room and the current presence state.
+ * It will use the instance belonging to the room in the nearest {@link RoomProvider} in the component tree.
+ * On calling, the hook will subscribe to the presence state of the room and update the state accordingly.
+ *
+ * @param params - Allows the registering of optional callbacks.
+ * @returns UsePresenceResponse - An object containing the {@link Presence} instance and the current presence state.
+ */
+export const usePresenceListener = (params?: UsePresenceListenerParams): UsePresenceListenerResponse => {
+  const { currentStatus: connectionStatus, error: connectionError } = useChatConnection({
+    onStatusChange: params?.onConnectionStatusChange,
+  });
+  const { room, roomError, roomStatus } = useRoom({
+    onStatusChange: params?.onRoomStatusChange,
+  });
+
+  const receivedEventNumber = useRef(0);
+  const triggeredEventNumber = useRef(0);
+  const retryTimeout = useRef<NodeJS.Timeout>();
+  const numRetries = useRef(0);
+  const latestPresentData = useRef<PresenceMember[]>([]);
+
+  useEffect(() => {
+
+    // handle the received event, ensuring we only process and return the latest presence data.
+    const handleReceivedEvent = () => {
+      receivedEventNumber.current += 1;
+
+      // clear the previous retry if we have received a new event
+      if (retryTimeout.current) {
+        clearTimeout(retryTimeout.current);
+        retryTimeout.current = undefined;
+        numRetries.current = 0;
+      }
+
+      // attempt to get the presence data
+      getAndEmit(receivedEventNumber.current);
+    };
+
+    const getAndEmit = (eventNumber: number) => {
+      room.presence
+        .get({waitForSync: true})
+        .then((presenceMembers) => {
+          // clear the retry now we have resolved
+          if (retryTimeout.current) {
+            clearTimeout(retryTimeout.current);
+            retryTimeout.current = undefined;
+            numRetries.current = 0;
+          }
+
+          // ensure the current event is still the latest
+          if (triggeredEventNumber.current >= eventNumber) {
+            return;
+          }
+
+          triggeredEventNumber.current = eventNumber;
+
+          // check if the presence data has changed, avoid unnecessary re-renders
+          if (dequal(latestPresentData.current, presenceMembers)) {
+            return;
+          }
+
+          // update the presence data
+          latestPresentData.current = presenceMembers;
+        })
+        .catch(() => {
+          const willReattempt = numRetries.current < PRESENCE_GET_MAX_RETRIES;
+
+          if (!willReattempt) {
+            console.log('Failed to fetch presence data after max retries');
```

Comment:
It would be good, at some point, to have our hooks write to the actual logger in the `ChatClient`.

We should do a hook (internal only) that does `useLogger` or similar, to access the chat clients logger.

Can we create a ticket to look into that post release?